### PR TITLE
Add option to toggle dislike submission in popup

### PR DESCRIPTION
**Pull request by VidithM**
In response to #292

### CODE REVIEW

**Review by glenlalo — COMMENTED** [review]
The button state doesn't get saved and returns to its original state each time I open the extension. 
grafik

File: Extensions/combined/popup.js [hunk 1]
```diff
@@ -29,6 +29,40 @@ document.getElementById("link_donate").addEventListener("click", () => {
   chrome.tabs.create({ url: config.links.donate });
 });
 
+loadSubmissionToggle();
+
+/*   Dislike submission toggle   */
+function loadSubmissionToggle(){
+  document.getElementById("dislike_submission_toggle").style.visibility = "hidden";
```
Comment (glenlalo) [inline Extensions/combined/popup.js:36]: I think that this line can be removed.

**Comment by mphelp** [comment]
@VidithM @glenlalo 
You need to leverage browser storage sync in order to persist the setting state.

**Comment by VidithM** [comment]
@glenlalo @mphelp Updated the UI and integrated storage sync. Let me know how it works.

File: Extensions/combined/popup.js [hunk 2]
```diff
@@ -29,6 +29,29 @@ document.getElementById("link_donate").addEventListener("click", () => {
...
+    slider.addEventListener("change", () => {
+      let state = (document.getElementById("dislike_submission_toggle").checked);
+      //alert(document.getElementById("dislike_submission_toggle").checked + " " + state);
+      chrome.storage.sync.set({"do_submission" : state});
+    });
+  });
+}
+
+function setSubmissionOptions(currVal){
+  //alert('Set submission opts with ' + currVal);
```
Comment (leonbubova) [inline Extensions/combined/popup.js:52]: Remove debug code

File: Extensions/combined/popup.js [hunk 3]
```diff
@@ -29,6 +29,29 @@ document.getElementById("link_donate").addEventListener("click", () => {
...
+  let slider = document.getElementById("dislike_submission_toggle");
+  slider.style.visibility = "hidden";
+  chrome.storage.sync.get(["do_submission"], (resp) => {
+    let val = resp["do_submission"];
+    //alert(val);
+    setSubmissionOptions(val);
+    slider.style.visibility = "visible";
+    slider.addEventListener("change", () => {
+      let state = (document.getElementById("dislike_submission_toggle").checked);
+      //alert(document.getElementById("dislike_submission_toggle").checked + " " + state);
```
Comment (leonbubova) [inline Extensions/combined/popup.js:45]: Remove debug code

File: Extensions/combined/popup.js [hunk 4]
```diff
@@ -29,6 +29,29 @@ document.getElementById("link_donate").addEventListener("click", () => {
...
 
+loadSubmissionToggle();
+
+/*   Dislike submission toggle   */
+function loadSubmissionToggle(){
+  let slider = document.getElementById("dislike_submission_toggle");
+  slider.style.visibility = "hidden";
+  chrome.storage.sync.get(["do_submission"], (resp) => {
+    let val = resp["do_submission"];
+    //alert(val);
```
Comment (leonbubova) [inline Extensions/combined/popup.js:40]: Remove debug code

File: Extensions/combined/ryd.content-script.js [hunk 5]
```diff
@@ -155,10 +156,16 @@ function RYD() {
...
+    RYDTools.getBrowser().storage.sync.get(["do_submission"], (resp) => {
+      if(!resp["do_submission"]){
+        return;
+      }
+      console.log("Sending vote...");
+      RYDTools.getBrowser().runtime.sendMessage({
+        message: "send_vote",
+        vote: vote,
+        videoId: getVideoId(window.location.href)
+      });
```
Comment (leonbubova) [inline Extensions/combined/ryd.content-script.js:168]: Why not make the if-clause check for the positive outcome and just let out the negative part all together.

```suggestion
    RYDTools.getBrowser().storage.sync.get(["do_submission"], (resp) => {
      if(resp["do_submission"]){
        console.log("Sending vote...");
        RYDTools.getBrowser().runtime.sendMessage({
          message: "send_vote",
          vote: vote,
          videoId: getVideoId(window.location.href)
        });
      }
```

Comment (leonbubova) [reply Extensions/combined/ryd.content-script.js:168]: Also you should think about removing the console.log's. Once you start keeping those in for prod it soon starts to get out of hand imo

File: Extensions/combined/ryd.content-script.js [hunk 6]
```diff
@@ -365,7 +372,14 @@ function RYD() {
     window.returnDislikeButtonlistenersSet = false;
```
Comment (leonbubova) [inline Extensions/combined/ryd.content-script.js:372]:
```suggestion
    window.returnDislikeButtonListenersSet = false;
```

File: Extensions/combined/ryd.content-script.js [hunk 7]
```diff
@@ -365,7 +372,14 @@ function RYD() {
...
     setEventListeners();
   });
-
+  RYDTools.getBrowser().storage.get(["init_submission"], (resp) => {
+    if(resp.length == 0){
```
Comment (leonbubova) [inline Extensions/combined/ryd.content-script.js:376]: You should either check with the typesafe `===` operator or just do something like
```
if(resp){ ...
```

File: Extensions/combined/ryd.content-script.js [hunk 8]
```diff
@@ -365,7 +372,14 @@ function RYD() {
...
+      RYDTools.getBrowser().storage.sync.set({"do_submission": true}, () => {
+        console.log("Initializd do_submission");
```
Comment (leonbubova) [inline Extensions/combined/ryd.content-script.js:378]: Same as above for the console.log

**Comment by leonbubova** [comment]
Hey there, I just did some testing and everything seemed to be working fine, but it appears that the default setting after installing is, to have submission of votes disabled. Could you maybe double check that and set it to be on by default if it's the same for you? I would say that makes sense here.

Also could you make a tiny ui change to the toggle button?
image
Right now it's cramped directly under the text. If you could give it a bit space that would be awesome.

**Comment by VidithM** [comment]
@leonbubova Should be good now.

**Comment by sy-b** [comment]
A problem:

What if a user
1. dislikes with dislike submission "on",
2. un-dislikes in "off" mode
3. again dislikes in "on" mode

**Comment by VidithM** [comment]
@sy-b I believe every vote is sent with the user ID saved in the browser storage, so this doesn't look like a problem

**Review by leonbubova — APPROVED** [review]

**Comment by Anarios** [comment]
Sorry, this was done in #326 as well.

**Pull request closed without merging**